### ISSUE TITLE
Location.getCurrentPositionAsync hangs indefinetly

### ISSUE THREAD

**Issue by BinaryVigilante** [expo/expo#15478]
### Summary

The **Location.getCurrentPositionAsync** just hangs indefinetly. I copied the code from the docs (https://docs.expo.dev/versions/latest/sdk/location/) and have a file called MapView.js with the following code:
```js
import React, { useState, useEffect } from 'react';
import { Text, View } from 'react-native';
import * as Location from 'expo-location';

export default function App() {
  const [location, setLocation] = useState(null);
  const [errorMsg, setErrorMsg] = useState(null);

  useEffect(() => {
    (async () => {
      let { status } = await Location.requestForegroundPermissionsAsync();
      if (status !== 'granted') {
        setErrorMsg('Permission to access location was denied');
        return;
      }

      let location = await Location.getCurrentPositionAsync({});
      //Doesn't continue after this line
      setLocation(location);
    })();
  }, []);

  let text = 'Waiting..';
  if (errorMsg) {
    text = errorMsg;
  } else if (location) {
    text = JSON.stringify(location);
  }

  return (
    <View>
      <Text>{text}</Text>
    </View>
  );
}
```
When I run the code in the app it's stuck on "Waiting.." and I'm getting no errors in the emulator nor the console. The issue is both on the emulator but also physical devices, I have both Nokia 4.2 and a Samsung A5.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### Environment

Expo CLI 4.13.0 environment info:
    System:
      OS: Linux 5.15 Ubuntu 21.10 21.10 (Impish Indri)
      Shell: 5.1.8 - /bin/bash
    Binaries:
      Node: 16.13.1 - /snap/bin/node
      Yarn: 1.22.15 - /snap/bin/yarn
      npm: 8.1.2 - /snap/bin/npm
    npmPackages:
      expo: ^43.0.0 => 43.0.0 
      react: 17.0.1 => 17.0.1 
      react-native: ^0.64.3 => 0.64.3 
      react-navigation: ^4.4.4 => 4.4.4 
    Expo Workflow: bare

### Reproducible demo

The code is identical to the code on the docs: https://docs.expo.dev/versions/latest/sdk/location/

**Comment by brentvatne** [comment]
there is more to just copying in code when it comes to a reproducible example. if you can create a new issue with a minimal reproducible example we can look but this is otherwise not actionable

**Comment by BinaryVigilante** [comment]
Please re-open this. This is the reproducible example. It is as minimum as it gets. This code is directly taken from the documentation and is supposed to work as is. I have not just copying in code, this code is in my app.

With the code from the documentation, `Location.getCurrentPositionAsync` hangs indefinitely.

**Comment by bradjones1** [comment]
I am experiencing this as well; either the documentation is incorrect (which is a docs bug) or there's an issue with the method, but concur that the code in the docs _is_ the minimal reproducible example.

**Comment by bradjones1** [comment]
I can reproduce this on an AVD emulator, but it works on a "real" device.

**Comment by marinoandrea** [comment]
I am encountering the same issue. However, as @bradjones1 said, this only appears to affect Android emulators while it works fine on real devices.
I have made a simple snack that reproduces the hanging behavior available at:

https://snack.expo.dev/@andrea-marino/issue_location_getcurrentlocationasync

If you run this on an emulated device (e.g. using the device ID) you can see how the 'Loading' text never changes to 'Done' and the location is never displayed.

**Comment by mikeRChambers610** [comment]
I am still having this issue and I am on Expo managed app SDK 47. getCurrentPositionAsync hangs indefinitely with no response. Event using lowest accuracy for android and ios.

**Comment by IordanisSap** [comment]
I am also getting this in a real android device

**Comment by sharmanderosier-rbi** [comment]
This is still and issue and is impacting real devices.

**Comment by melissa07** [comment]
I'm experiencing this as well on a real Android device.

**Comment by da45** [comment]
I found the same issue,  I am stuck here, any fix please?

**Comment by shahzaib-cmyk** [comment]
Facing the same issue here on SDK 49.0.6, the issue also happens on real devices, both on dev-client and preview builds on other Android devices. I'm only working on Android btw. Devs on another issue suggested using timeouts to retry the function 3-4 times and for some, it worked, however for me, it hangs indefinitely on every retry. This issue popped up after I upgraded the SDK version from 46 all the way to 49 (Somewhere along the way, this probably happened). I still have a working version of the app that uses "expo-location": "~14.3.0", but the newer "expo-location": "~16.1.0" hangs indefinitely for me. Even `watcwatchPositionAsync` hangs 😭.

**Comment by bradchristensen** [comment]
Like @shahzaib-cmyk, this issue started occurring in our app (on real devices running the production app build) only after upgrading the following:

- React Native from `0.70.6` to `0.72.4`
- `expo@47.0.6` to `expo@49.0.8`
- `expo-location@15.0.1` to `expo-location@16.1.0`

Our error logging has picked the issue up on several versions of Android (10, 12, 13) and at least iOS 16.6.1.

We've also started working around it by implementing 10-second timeouts/retries and falling back to fetching low accuracy location - **the low accuracy fallback seems to work very reliably** and I think we've only seen that fail once, while we've seen hundreds of timeouts for high-accuracy  or 'last known location'.

I would expect timeouts to be built into the `getCurrentPositionAsync` API, with a sensible default, as it doesn't make sense for it to hang indefinitely.

I'm curious to know what's caused this to happen only now, and why it wasn't an issue before. Maybe it was automatically falling back to a low accuracy value previously, and the new behaviour is more correct (albeit unexpected, from my point of view)?

**Comment by pgeimer** [comment]
Having the same issue on Android 13. We are using `expo@49.0.13` and `expo-location@16.1.0`. 
Usually the freeze happens on the first call of `getCurrentPositionAsync()` after app start. We came up with a second retry after 10 seconds.

```typescript
async function getLocationWithRetry(retries = 2): Promise<Location.LocationObject> {
    const timeout = new Promise<never>((_, reject) =>
        setTimeout(() => reject(new Error('Timeout exceeded')), 10000)
    );
    try {
        return await Promise.race([Location.getCurrentPositionAsync({}), timeout]) as Location.LocationObject;
    } catch (error) {
        if (retries > 0) {
            return getLocationWithRetry(retries - 1);
        } else {
            throw error;
        }
    }
}

const userLocation = await getLocationWithRetry();
```
Its working for us but its not a fine solution.

**Comment by nicholas-masini** [comment]
I have the same issue on Android 13 as well. The hanging is inconsistent, sometimes it works immediately but most of the time it just hangs getting the location. I have also implemented a retry mechanism for this functionality but on Android it still just hangs indefinitely even after a countless amount of retries. The accuracy I am using is Balanced. Seems quite inconvenient that the Location API does not handle this internally

**Comment by shahzaib-cmyk** [comment]
> Facing the same issue here on SDK 49.0.6, the issue also happens on real devices, both on dev-client and preview builds on other Android devices. I'm only working on Android btw. Devs on another issue suggested using timeouts to retry the function 3-4 times and for some, it worked, however for me, it hangs indefinitely on every retry. This issue popped up after I upgraded the SDK version from 46 all the way to 49 (Somewhere along the way, this probably happened). I still have a working version of the app that uses "expo-location": "~14.3.0", but the newer "expo-location": "~16.1.0" hangs indefinitely for me. Even `watcwatchPositionAsync` hangs 😭.

Update: I figured out what I was doing wrong. I was using my mobile hotspot on the phone running the dev-client and had it connected to the laptop with the dev-server. I had to do this because the wifi was acting up, but this was somehow preventing me from getting the user's location. Just in case someone faces the same issue, try connecting with a different network.

**Comment by andrecrimb** [comment]
> the location service only works in emulators if you are logged into a google account.

Solved the emulator issue 👍🏽

**Comment by Developeranees** [comment]
same issue on sdk 49

**Comment by Noitidart** [comment]
> same issue on sdk 49

@Developeranees were you able to solve this? Im having same issue on Expo 49

**Comment by Mikewa33** [comment]
Still seeing this issue on emulator at least
